### PR TITLE
Feature/burn block time

### LINKED ISSUE #949: Fill in block-time stub in block-info
### Motivation

Make absolute time easier to work with in smart contracts. 

The current method for estimating the absolute time requires hardcoding in a timestamp during contract deployment. For example:

```scm
(define height-at-launch block-height)
(define time-at-launch 1557237920)
(define block-time (* 60 10))

(define (estimate-current-time)
  (+ time-at-launch 
    (* (- block-height height-at-launch) block-time)))
```

### Implementation

Expose the bitcoin median timestamp as a built in variable, similar to `block-height`. Should be explicitly named so that it is immediately clear that the timestamp is not realtime, for example `median-time`. 

### Usage reliability

Bitcoin defines median time as:
> The median time of the 11 blocks before the most recent block on the blockchain. Used for validating transaction locktime under [BIP113](https://github.com/bitcoin/bips/blob/master/bip-0113.mediawiki).

BIP113 states that it ensures this median time "increases monotonically with each block". 

> The block time is a Unix epoch time when the miner started hashing the header (according to the miner). Must be strictly greater than the median time of the previous 11 blocks. Full nodes will not accept blocks with headers more than two hours in the future according to their clock.

See here for block timestamp validation src: https://github.com/bitcoin/bitcoin/blob/0.18/src/validation.cpp#L3258

The `mediantime` is included in several bitcoin RPC methods ([`getblock`](https://bitcoincore.org/en/doc/0.18.0/rpc/blockchain/getblock/), [`getblockchaininfo`](https://bitcoincore.org/en/doc/0.18.0/rpc/blockchain/getblockchaininfo/), [`getblockstats`](https://bitcoincore.org/en/doc/0.18.0/rpc/blockchain/getblockstats), etc).

Additionally, lightning network depends on this variable for CTLV timelocks https://github.com/lightningnetwork/lnd/search?p=2&q=CLTV&unscoped_q=CLTV

### PR DESCRIPTION
**Pull request by jcnelson**
Un-stub burnchain block time in `get-block-info`.  Fixes #949 and #1131

### CODE REVIEW

File: src/core/mod.rs [hunk 1]
```diff
@@ -24,7 +24,7 @@ use chainstate::burn::BlockHeaderHash;
 use util::log;
 
 // fork set identifier -- to be mixed with the consensus hash (encodes the version)
-pub const SYSTEM_FORK_SET_VERSION : [u8; 4] = [22u8, 0u8, 0u8, 0u8];
+pub const SYSTEM_FORK_SET_VERSION : [u8; 4] = [23u8, 0u8, 0u8, 0u8];
 
 // p2p network version 
 pub const PEER_VERSION : u32 = 0x16000000;      // 22.0.0.0
```
Comment (kantai) [inline src/core/mod.rs]: should this increment with `SYSTEM_FORK_SET_VERSION`?

File: src/chainstate/stacks/db/blocks.rs [hunk 2]
```diff
@@ -173,17 +175,29 @@ impl FromRow<StagingBlock> for StagingBlock {
         let sortition_burn_i64 : i64 = row.get("sortition_burn");
         let block_data : Vec<u8> = vec![];
 
+        if commit_burn_i64 < 0 {
```
Comment (kantai) [inline src/chainstate/stacks/db/blocks.rs]: Do you think it's worthwhile for us to implement `FromSql<u64>` and `ToSql<u64>` ? It'd reduce some of the repetitive error handling for getting u64s into and out of the database.

Comment (jcnelson) [reply src/chainstate/stacks/db/blocks.rs]: Yes, thanks.

File: src/core/mod.rs (hunk 1 above)
Comment (jcnelson) [reply src/core/mod.rs]: Yes, thanks.

File: src/chainstate/stacks/db/blocks.rs (hunk 2 above)
Comment (jcnelson) [reply src/chainstate/stacks/db/blocks.rs]: Will do this in a separate PR: https://github.com/blockstack/blockstack-core/issues/1260

**Pull request merged**